packages/flame/lib/src/components/core/component_tree_root.dart

```diff
@@ -127,6 +127,23 @@ class ComponentTreeRoot extends Component {
       }
     });
   }
+
+  @mustCallSuper
+  @internal
+  void registerKey(ComponentKey key, Component component) {
+    _index[key] = component;
```

Comment:
Maybe put an assertion here if the key is already in the map?

Reply:
Done!